src/eko/matching_conditions/nnlo.py

```diff
@@ -127,10 +162,14 @@ def A_hg_2(n, sx):
     S1 = sx[0]
     S2 = sx[1]
     S3 = sx[2]
+    S1m = harmonic_S1(n - 1)
+    S2m = harmonic_S2(n - 1)
+    S2mhalf = harmonic_S2((n - 1) / 2)
```

Comment:
- if you wish you can call this `Sp2m` https://github.com/N3PDF/eko/blob/feature/backward_evol/src/eko/anomalous_dimensions/nlo.py#L41 and `Sp2p`
- we should finally start caching the S-es also in `ad.nlo` and then reuse them here

Reply:
>     we should finally start caching the S-es also in ad.nlo and then reuse them here

actually that's a bit stupid, because `ad` and `matching` are never computed in the same run and thus could share S-es ...